### ISSUE TITLE
"Save as ..." actions don't respect result set sorting

### ISSUE THREAD

**Issue by Charles-Gagnon** [microsoft/azuredatastudio#19030]
1. Run query with results
2. Sort results so they're in a different order than the initial results
3. Use one of the "Save as" actions (such as "Save as JSON")

The file is generated, but the items are in the initial order. They should be in the displayed order. 

Likely similar issue as https://github.com/microsoft/azuredatastudio/pull/18973

**Comment by Charles-Gagnon** [comment]
@alanrenmsft Assigning this to you for triaging

**Comment by alanrenmsft** [comment]
this is not a regression, when filtering/sorting is introduced, I didn't implement what you have described. It is fair to ask for this feature, but on the other hand, the filtering/sorting are just on manipulating the client side data, when serialized, user may want all the data.

putting this in backlog and if see user votes we can revisit it.

**Comment by Charles-Gagnon** [comment]
This is more noticeable when you select items and export those. Instead of the items you've selected you get completely different items. I'd definitely consider that functionality a bug.

**Comment by alanrenmsft** [comment]
agreed, the select items and export scenario is confusing, let me move this to June release and take a look.
